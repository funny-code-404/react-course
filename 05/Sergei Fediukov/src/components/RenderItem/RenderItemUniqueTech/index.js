import { UniqueTech } from '../../../pages'
import React from 'react'
class RenderItemUniqueTech extends React.Component {
    state = {
        unique_tech: {
            0: false,
            1: false
        },
    }
    render() {
        const unique_tech = this.props.unique_tech
        const hendleUniqueTech = (e) => {
            const i = e.target.attributes[0].value
            this.setState(prevState => ({
                ...prevState,
                unique_tech: {
                    ...prevState.unique_tech,
                    [i]: !prevState.unique_tech[i]
                }
            }))
        }
        return (
            <div>
                <p>Unique tech:</p>
                <ul>
                    {unique_tech.map((x, i) =>
                        <li key={x}>
                            <div>
                                <button onClick={hendleUniqueTech} i={i}>
                                    {
                                        !this.state.unique_tech[i] ?
                                            x.split('/')[x.split('/').length - 1].replace('_', ' ')
                                            : 'close'
                                    }
                                </button>
                            </div>
                            {
                                this.state.unique_tech[i] &&
                                <div>
                                    <UniqueTech url={x} />
                                </div>
                            }
                        </li>)}
                </ul>
            </div>
        )
    }
}
export default RenderItemUniqueTech

