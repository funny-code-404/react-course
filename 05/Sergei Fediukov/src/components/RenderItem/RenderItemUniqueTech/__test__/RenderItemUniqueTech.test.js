import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemUniqueTech from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {
    const mockFn = jest.fn();
    const props = {
        unique_tech: ['test tech'],
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<RenderItemUniqueTech {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component has children button', () => {
        expect(wrapper.find('button')).toHaveLength(1)
    })

})