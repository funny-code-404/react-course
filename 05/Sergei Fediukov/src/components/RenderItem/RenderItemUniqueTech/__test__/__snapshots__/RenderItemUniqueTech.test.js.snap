// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div>
  <p>
    Unique tech:
  </p>
  <ul>
    <li
      key="test tech"
    >
      <div>
        <button
          i={0}
          onClick={[Function]}
        >
          test tech
        </button>
      </div>
    </li>
  </ul>
</div>
`;
